File: src/php/core/src/Style/Meta.php
<?php

declare(strict_types=1);

namespace DiceBear\Style;

/**
 * Lazily-constructed view over a style definition's `meta` block, exposing
 * the license, creator, and source descriptors.
 */
class Meta
{
    private ?MetaLicense $license = null;
    private ?MetaCreator $creator = null;
    private ?MetaSource $source = null;

    /**
     * @param array<string, mixed> $data
     */
    public function __construct(private readonly array $data) {}

    /**
     * Returns the license descriptor, defaulting to an empty object when the
     * style definition omits the field.
     */
    public function license(): MetaLicense
    {
        return $this->license ??= new MetaLicense($this->data['license'] ?? []);
    }

    /**
     * Returns the creator descriptor, defaulting to an empty object when the
     * style definition omits the field.
     */
    public function creator(): MetaCreator
    {
        return $this->creator ??= new MetaCreator($this->data['creator'] ?? []);
    }

    /**
     * Returns the source descriptor, defaulting to an empty object when the
     * style definition omits the field.
     */
    public function source(): MetaSource
    {
        return $this->source ??= new MetaSource($this->data['source'] ?? []);
    }
}

File: src/rust/core/src/style/meta.rs
//! Lazily-constructed view over a style definition's `meta` block.

use serde::Deserialize;

use super::meta_creator::MetaCreator;
use super::meta_license::MetaLicense;
use super::meta_source::MetaSource;

#[derive(Deserialize, Default)]
pub(crate) struct Meta {
    #[serde(default)]
    license: MetaLicense,
    #[serde(default)]
    creator: MetaCreator,
    #[serde(default)]
    source: MetaSource,
}

impl Meta {
    pub(crate) fn license(&self) -> &MetaLicense {
        &self.license
    }

    pub(crate) fn creator(&self) -> &MetaCreator {
        &self.creator
    }

    pub(crate) fn source(&self) -> &MetaSource {
        &self.source
    }
}
